app/src/main/java/com/islaharper/dawnofandroid/presentation/screens/login/LoginScreen.kt

```diff
@@ -23,69 +29,118 @@ import androidx.compose.ui.tooling.preview.Preview
 import androidx.compose.ui.unit.dp
 import androidx.hilt.navigation.compose.hiltViewModel
 import androidx.navigation.NavHostController
+import com.google.android.gms.auth.api.identity.BeginSignInResult
+import com.google.android.gms.common.api.ApiException
 import com.islaharper.dawnofandroid.R
 import com.islaharper.dawnofandroid.domain.model.MessageBarState
+import com.islaharper.dawnofandroid.navigation.Screen
 import com.islaharper.dawnofandroid.presentation.common.GoogleButton
 import com.islaharper.dawnofandroid.presentation.components.MessageBar
+import com.islaharper.dawnofandroid.ui.theme.DawnOfAndroidTheme
+import com.islaharper.dawnofandroid.util.Resource
 
 @Composable
 fun LoginScreen(
     navHostController: NavHostController,
-    loginViewModel: LoginViewModel = hiltViewModel()
+    modifier: Modifier = Modifier,
+    loginViewModel: LoginViewModel = hiltViewModel(),
 ) {
-    val signedInState by loginViewModel.signedInState.collectAsState()
+    val launchOneTapState by loginViewModel.launchOneTapSignIn.collectAsState()
     val messageBarState by loginViewModel.messageBarState.collectAsState()
-    val apiResponse by loginViewModel.apiResponse.collectAsState()
+    val navigationState by loginViewModel.navigationState.collectAsState()
+    val oneTapSignInResponse by loginViewModel.oneTapSignInResponse.collectAsState()
 
     Scaffold(
         backgroundColor = MaterialTheme.colorScheme.surface,
         content = { contentPadding ->
             LoginContent(
-                modifier = Modifier.padding(contentPadding),
-                signedInState = signedInState,
+                modifier = modifier.padding(contentPadding),
+                signedInState = launchOneTapState,
                 messageBarState = messageBarState,
                 onButtonClicked = {
-                    loginViewModel.saveSignedInState(signedIn = true)
-                }
+                    loginViewModel.onGoogleButtonClick()
+                },
             )
-        }
+        },
     )
+
+    val authLauncher = authLauncher(loginViewModel)
```

Comment:
I don't think that extracting a function helps here, especially when the function is a Composable and the parameter `ViewModel` is not stable. The complexity that we seem to be trying to hide via extracting the function is better solved another way. In this case, by removing the try/catch. If the try/catch is necessary, can you please attach a screenshot where the debugger has stopped inside the catch block and copy/paste the stacktrace?

```suggestion
    val authLauncher = rememberLauncherForActivityResult(
        ActivityResultContracts.StartIntentSenderForResult(),
    ) { result ->
        if (result.resultCode == Activity.RESULT_OK) {
            loginViewModel.onOneTapSignInSuccess(result = result.data)
        } else {
            loginViewModel.onOneTapSignInFailure(failureMessage = "Error Signing In")
        }
    }
```